production/tanka/grafana-agent/grafana-agent.libsonnet

```diff
@@ -53,4 +54,67 @@ k + config {
     daemonSet.mixin.spec.template.spec.withServiceAccount($._config.agent_cluster_role_name) +
     config_hash_mixin +
     $.util.configVolumeMount($._config.agent_configmap_name, '/etc/agent'),
+
+
+  local agent_deployment_configmap_name = $._config.agent_configmap_name + '_deployment',
+  // If running on GKE, you cannot scrape API server pods, and must
+  // instead scrape the API server service endpoints. On AKS this doesn't
+  // work.
+  agent_deployment_config_map:
+    if $._config.agent_host_filter then
+      configMap.new(agent_deployment_configmap_name) +
+      configMap.withData({
+        'agent.yml': $.util.manifestYaml($._config.agent_config {
+          configs: [{
+            name: 'agent',
+            host_filter: $._config.agent_host_filter,
+            remote_write: $._config.agent_remote_write,
+            scrape_configs: [{
+              job_name: 'default/kubernetes',
+              kubernetes_sd_configs: [{
+                role:
+                  if $._config.scrape_api_server_endpoints
+                  then 'endpoints'
+                  else 'service',
+              }],
+              scheme: 'https',
+
+              tls_config: {
+                ca_file: '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
+                insecure_skip_verify: $._config.prometheus_insecure_skip_verify,
+              },
+              bearer_token_file: '/var/run/secrets/kubernetes.io/serviceaccount/token',
+
+              relabel_configs: [{
+                source_labels: ['__meta_kubernetes_service_label_component'],
+                regex: 'apiserver',
+                action: 'keep',
+              }],
+
+              // Drop some high cardinality metrics.
+              metric_relabel_configs: [
+                {
+                  source_labels: ['__name__'],
+                  regex: 'apiserver_admission_controller_admission_latencies_seconds_.*',
+                  action: 'drop',
+                },
+                {
+                  source_labels: ['__name__'],
+                  regex: 'apiserver_admission_step_admission_latencies_seconds_.*',
+                  action: 'drop',
+                },
+              ],
+            }],
+          }],
+        }),
+      })
+    else {},
+
+  agent_deployment:
```

Comment:
Can this add in the `config_hash_mixin`?

Reply:
done!